lib/scholar/manifold/tsne.ex

```diff
@@ -167,18 +167,21 @@ defmodule Scholar.Manifold.TSNE do
       |> Nx.reshape({num_samples, 1, num_features})
       |> Nx.broadcast(broadcast_shape)
 
-    case metric do
+    case opts[:metric] do
       :squared_euclidean ->
         Distance.squared_euclidean(t1, t2, axes: [2])
+        # Distance.pairwise_squared_euclidean(x)
 
       :euclidean ->
         Distance.euclidean(t1, t2, axes: [2])
+        # Distance.pairwise_euclidean(x)
 
       :manhattan ->
         Distance.manhattan(t1, t2, axes: [2])
 
       :cosine ->
         Distance.cosine(t1, t2, axes: [2])
+        # Distance.pairwise_cosine(x)
```

Comment:
I have no idea why using pairwise distances causes the results to be completely different than those from broadcasting and calculating distances by hand. I tested this function, and it returns the same results for the same input value.

Reply:
Try doing `print_expr` and see if the expressions are the same or different?

Reply:
Okay, I discovered that after 10 iterations of tsne algorithms, both functions gives the same results. After 50, there is a small discrepancy in distances, and after 500 iterations (default) there are completely different results.